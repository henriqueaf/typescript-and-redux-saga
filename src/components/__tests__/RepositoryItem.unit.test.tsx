import React from 'react';
import * as TestRenderer from 'react-test-renderer';

import { Repository } from '../../store/ducks/repositories/types';
import RepositoryItem from '../RepositoryItem';

describe('RepositoryItem', () => {
  it('Renders a list item with a repository name', () => {
    const repository: Repository = {
      id: 1,
      name: 'teste',
    };

    const wrapper = TestRenderer.create(
      <RepositoryItem repository={repository} />
    );

    const repositoryItemInstance = wrapper.root;

    expect(repositoryItemInstance.findByType('li').children[0]).toBe('teste');
    expect(repositoryItemInstance).toMatchSnapshot();
  });
});
