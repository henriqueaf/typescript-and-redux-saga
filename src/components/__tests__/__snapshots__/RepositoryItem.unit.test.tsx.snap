// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryItem Renders a list item with a repository name 1`] = `
ReactTestInstance {
  "_fiber": FiberNode {
    "_debugHookTypes": null,
    "_debugID": 3,
    "_debugIsCurrentlyTiming": false,
    "_debugNeedsRemount": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "/home/henrique/reactjs_workspace/typesript-react/src/components/__tests__/RepositoryItem.unit.test.tsx",
      "lineNumber": 15,
    },
    "actualDuration": 0,
    "actualStartTime": -1,
    "alternate": null,
    "child": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 4,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "/home/henrique/reactjs_workspace/typesript-react/src/components/RepositoryItem/index.tsx",
        "lineNumber": 8,
      },
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": null,
      "child": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 5,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": null,
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": "teste",
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": "teste",
        "ref": null,
        "return": [Circular],
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": Object {
          "isHidden": false,
          "tag": "TEXT",
          "text": "teste",
        },
        "tag": 6,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": null,
      },
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": "li",
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": "teste",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": "teste",
      },
      "ref": null,
      "return": [Circular],
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": Object {
        "children": Array [
          Object {
            "isHidden": false,
            "tag": "TEXT",
            "text": "teste",
          },
        ],
        "isHidden": false,
        "props": Object {
          "children": "teste",
        },
        "rootContainerInstance": Object {
          "children": Array [
            [Circular],
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "tag": "INSTANCE",
        "type": "li",
      },
      "tag": 5,
      "treeBaseDuration": 0,
      "type": "li",
      "updateQueue": null,
    },
    "childExpirationTime": 0,
    "dependencies": null,
    "effectTag": 1,
    "elementType": [Function],
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "repository": Object {
        "id": 1,
        "name": "teste",
      },
    },
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {
      "repository": Object {
        "id": 1,
        "name": "teste",
      },
    },
    "ref": null,
    "return": FiberNode {
      "_debugHookTypes": null,
      "_debugID": 1,
      "_debugIsCurrentlyTiming": false,
      "_debugNeedsRemount": false,
      "_debugOwner": null,
      "_debugSource": null,
      "actualDuration": 0,
      "actualStartTime": -1,
      "alternate": FiberNode {
        "_debugHookTypes": null,
        "_debugID": 1,
        "_debugIsCurrentlyTiming": false,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "_debugSource": null,
        "actualDuration": 0,
        "actualStartTime": -1,
        "alternate": [Circular],
        "child": null,
        "childExpirationTime": 0,
        "dependencies": null,
        "effectTag": 0,
        "elementType": null,
        "expirationTime": 1073741823,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "selfBaseDuration": 0,
        "sibling": null,
        "stateNode": FiberRootNode {
          "callbackExpirationTime": 0,
          "callbackNode": null,
          "containerInfo": Object {
            "children": Array [
              Object {
                "children": Array [
                  Object {
                    "isHidden": false,
                    "tag": "TEXT",
                    "text": "teste",
                  },
                ],
                "isHidden": false,
                "props": Object {
                  "children": "teste",
                },
                "rootContainerInstance": [Circular],
                "tag": "INSTANCE",
                "type": "li",
              },
            ],
            "createNodeMock": [Function],
            "tag": "CONTAINER",
          },
          "context": Object {},
          "current": [Circular],
          "finishedExpirationTime": 0,
          "finishedWork": null,
          "firstBatch": null,
          "firstPendingTime": 0,
          "hydrate": false,
          "interactionThreadID": 1,
          "lastPendingTime": 0,
          "memoizedInteractions": Set {},
          "pendingChildren": null,
          "pendingContext": null,
          "pendingInteractionMap": Map {},
          "pingCache": null,
          "pingTime": 0,
          "tag": 0,
          "timeoutHandle": -1,
        },
        "tag": 3,
        "treeBaseDuration": 0,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "firstCapturedEffect": null,
          "firstCapturedUpdate": null,
          "firstEffect": null,
          "firstUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <_default
                repository={
                  Object {
                    "id": 1,
                    "name": "teste",
                  }
                }
              />,
            },
            "priority": 97,
            "suspenseConfig": null,
            "tag": 0,
          },
          "lastCapturedEffect": null,
          "lastCapturedUpdate": null,
          "lastEffect": null,
          "lastUpdate": Object {
            "callback": null,
            "expirationTime": 1073741823,
            "next": null,
            "nextEffect": null,
            "payload": Object {
              "element": <_default
                repository={
                  Object {
                    "id": 1,
                    "name": "teste",
                  }
                }
              />,
            },
            "priority": 97,
            "suspenseConfig": null,
            "tag": 0,
          },
        },
      },
      "child": [Circular],
      "childExpirationTime": 0,
      "dependencies": null,
      "effectTag": 0,
      "elementType": null,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <_default
          repository={
            Object {
              "id": 1,
              "name": "teste",
            }
          }
        />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "selfBaseDuration": 0,
      "sibling": null,
      "stateNode": FiberRootNode {
        "callbackExpirationTime": 0,
        "callbackNode": null,
        "containerInfo": Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "isHidden": false,
                  "tag": "TEXT",
                  "text": "teste",
                },
              ],
              "isHidden": false,
              "props": Object {
                "children": "teste",
              },
              "rootContainerInstance": [Circular],
              "tag": "INSTANCE",
              "type": "li",
            },
          ],
          "createNodeMock": [Function],
          "tag": "CONTAINER",
        },
        "context": Object {},
        "current": [Circular],
        "finishedExpirationTime": 0,
        "finishedWork": null,
        "firstBatch": null,
        "firstPendingTime": 0,
        "hydrate": false,
        "interactionThreadID": 1,
        "lastPendingTime": 0,
        "memoizedInteractions": Set {},
        "pendingChildren": null,
        "pendingContext": null,
        "pendingInteractionMap": Map {},
        "pingCache": null,
        "pingTime": 0,
        "tag": 0,
        "timeoutHandle": -1,
      },
      "tag": 3,
      "treeBaseDuration": 0,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <_default
            repository={
              Object {
                "id": 1,
                "name": "teste",
              }
            }
          />,
        },
        "firstCapturedEffect": null,
        "firstCapturedUpdate": null,
        "firstEffect": null,
        "firstUpdate": null,
        "lastCapturedEffect": null,
        "lastCapturedUpdate": null,
        "lastEffect": null,
        "lastUpdate": null,
      },
    },
    "selfBaseDuration": 0,
    "sibling": null,
    "stateNode": null,
    "tag": 0,
    "treeBaseDuration": 0,
    "type": [Function],
    "updateQueue": null,
  },
}
`;
